### ISSUE TITLE
Unable to wrap `stat`

### ISSUE THREAD

**Issue by melonedo** [JuliaInterop/Clang.jl#320]
This happened when I am wrapping GDAL.jl.
On linux, if I attemp to wrap this code:
```c
#include <sys/stat.h>
typedef struct stat mystat;
```
because `stat` is also a function, `Audit` pass can not be passed. But it seems that neither `CollectTopLevelNode` nor `CollectDependentSystemNode` supports filtering, and `Audit` pass can not be disabled.
Do we need to add filtering earlier to solve this?

**Comment by Gnimuc** [comment]
Is it feasible to use `@add_def stat` to skip? Anyway, we can not define a struct and a function with the same name.

**Comment by Gnimuc** [comment]
related: https://github.com/JuliaInterop/Clang.jl/issues/290 

IMHO, this is a bug and should be fixed in the upstream.

**Comment by melonedo** [comment]
> Is it feasible to use @add_def stat to skip?

Currently `resolve_dependency!` and `collect_dependent_system_nodes!` for `TypedefElaborated` do not respect `dag.ids_extra`. They only check `dag.tags`. I modified them and it seems good.

**Comment by Gnimuc** [comment]
Maybe `dag.ids_extra` needs to be checked firstly, so `@add_def` takes the highest priority.

**Comment by visr** [comment]
Just to make sure, this is not because stat is being excluded in this rewriter?
https://github.com/JuliaGeo/GDAL.jl/blob/d44dc737ec16a60d1edb81bb71e0cb5f0666f165/gen/wrap.jl#L71

**Comment by visr** [comment]
> IMHO, this is a bug and should be fixed in the upstream.

Just shooting here, but could this be related? https://github.com/OSGeo/gdal/issues/4123

**Comment by melonedo** [comment]
I think real C compilers probably should not error because of this. Here Clang.jl found `stat` to be a dependency and tries to pull it out from system headers. However, it can not distinguish `stat` (a function, not actually used) from `struct stat` and includes both, which causes sanity check to fail.
As for your case, I only tested on linux so probably more symbols (e.g. `_stat64`) need to be ignored and added back in prologue.jl.
I should have mentioned earlier that these checks are ignored because I disabled macros as whole, and `stat` and `tm` are special-cased.
